ts/src/pro/binance.ts

```diff
@@ -1149,7 +1151,10 @@ export default class binance extends binanceRest {
         } else {
             signature = this.hmac (this.encode (query), this.encode (this.secret), sha256);
         }
-        extendedParams['signature'] = signature;
+        if (isPrivate) {
```

Comment:
why are we calculating the signature before knowing if we're going to need it, this does not make sense